packages/core/src/utils/make-namespaced.ts

```diff
@@ -163,74 +163,75 @@ export function makeNamespacedInput(input: SolcInput, output: SolcOutput, solcVe
       }
     }
 
-    const modifiedSource = tryRemoveNonStructNatSpec(getModifiedSource(orig, modifications), solcVersion);
-
-    modifiedSources[sourcePath] = { ...source, content: modifiedSource };
+    modifiedSources[sourcePath] = { ...source, content: getModifiedSource(orig, modifications) };
   }
 
   return { ...input, sources: modifiedSources, settings: { ...input.settings, outputSelection: OUTPUT_SELECTION } };
 }
 
 /**
- * If we have the compiler version available and Slang is supported for the current platform and compiler version,
- * use Slang to parse and remove all NatSpec comments that do not precede a struct definition and return the modified content.
+ * Attempts to remove all NatSpec comments that do not precede a struct definition from the input source contents.
+ * Directly modifies the input source contents.
+ *
+ * If the solc version is not supported by the parser, the original content is kept.
+ *
+ * @param solcInput Solc input.
+ * @param solcVersion The version of the solc compiler that was originally used to compile the input.
+ */
+export async function trySanitizeNatSpec(solcInput: SolcInput, solcVersion: string) {
+  for (const [sourcePath, source] of Object.entries(solcInput.sources)) {
+    if (source.content !== undefined) {
+      solcInput.sources[sourcePath].content = await tryRemoveNonStructNatSpec(source.content, solcVersion);
+    }
+  }
+}
+
+/**
+ * If Slang is supported for the current compiler version, use Slang to parse and remove all NatSpec comments
+ * that do not precede a struct definition and return the modified content.
  *
  * Otherwise, return the original content.
  */
-function tryRemoveNonStructNatSpec(origContent: string, solcVersion: string | undefined): string {
-  const natSpecRemovals: Modification[] = [];
+async function tryRemoveNonStructNatSpec(origContent: string, solcVersion: string): Promise<string> {
+  if (solcVersion === undefined) {
+    return origContent;
+  }
 
-  if (solcVersion !== undefined && tryRequire('@nomicfoundation/slang') && slangSupportsVersion(solcVersion)) {
-    /* eslint-disable @typescript-eslint/no-var-requires */
-    const { Language } = require('@nomicfoundation/slang/language');
-    const { NonterminalKind, TerminalKind } = require('@nomicfoundation/slang/kinds');
-    const { TerminalNode } = require('@nomicfoundation/slang/cst');
-    const { isTrivia } = require('./slang/trivia');
-    /* eslint-enable @typescript-eslint/no-var-requires */
+  const { Parser } = await import('@nomicfoundation/slang/parser');
+  if (!Parser.supportedVersions().includes(solcVersion)) {
```

Comment:
Is this breaking compatibility with older versions of Solidity? Perhaps we should mention it in the Changelog

Reply:
On the contrary, this adds compatibility for a newer version of Solidity.  So I agree this can be in the changelog and can have a minor version bump.

Reply:
Yes, let's expand a bit in the changelog entry to make it clear we're extending compatibility